### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by sangwa7**
# Hello reviewer ✋🏾

**Below are the tasks implemented:**

- [ ]  allowed user to add like to post
- [ ]  create comment and post on behalf of current user
- [ ]  return first user from database
- [ ]  fix linter errors

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      format.html
+      format.json { render json: @posts }
+    end
+  end
+
+  def create
+    post = params[:post]
+    user = current_user
+
+    new_post = Post.new(post.permit(:title, :text))
```
Comment (EricMbouwe) [inline app/controllers/posts_controller.rb]: - [ ] Let's isolatte permitted params in a private method to decouple the create method from that.
```ruby
private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :create, :new, :show] do
+      resources :comments, only: [:create :new]
```
Comment (EricMbouwe) [inline config/routes.rb]: - [ ] You are missing a comma here😉

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def create
+    comment_author = current_user
+    post = Post.find(params[:post_id])
+    post_author = post.user
+
+    comment = params[:comment]
+
+    new_comment = Comment.new(comment.permit(:text))
```
Comment (EricMbouwe) [inline app/controllers/comments_controller.rb]: - [ ] As mentioned above, let's extract the permit to a private method. You can learn more aout the strong parameters in this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)
```ruby
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end
```

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @sangwa7, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- ⚡ Good implementation of new and create methods for both post and comment
- ⚡ PR
- ⚡ Commits

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 2 above)
Comment (ahmetbozaci) [reply config/routes.rb]: - [ ] Kindly fix add comma in after create route like other code reviewer said

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (ahmetbozaci) [reply app/controllers/posts_controller.rb]: - [ ] like the last code reviewer said kindly use another method for params and make it private.

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (ahmetbozaci) [reply app/controllers/comments_controller.rb]: - [ ] Kindly use comment params  method and make it private.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️ 


@sangwa7  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___

### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._


___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
-  I cant create a database, or migrate the database. If it is like that in your side too kindly fix it.
image

___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by ahmetbozaci** [comment]
@sangwa7  I finish the review and you push something. Kindly don't work on the project after sending it for review.
If something you fix and ı review please ignore it.

**Review by Lameck1 — APPROVED** [review]
Hi, @sangwa7,

You've done well working on the concerns of the previous review 👍🏼.
The tasks for this milestone are now successfully implemented 💯.

## Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Please, remember to tag me in your question so I can receive the notification.**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**